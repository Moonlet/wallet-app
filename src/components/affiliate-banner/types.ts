export const AffiliateBannerType = {
    LEDGER_NANO_X: {
        url: 'https://shop.ledger.com/pages/ledger-nano-x?r=69d534ac55d9',
        image: require('../../assets/images/png/affiliate-banners/ledger_nano_x.png')
    },
    UNSTOPPABLE_DOMAINS: {
        url: 'https://unstoppabledomains.com/r/moonlet',
        image: require('../../assets/images/png/affiliate-banners/unstoppabledomains.png')
    }
};

export interface IAffiliateBannerType {
    url: string;
    image: any;
}
