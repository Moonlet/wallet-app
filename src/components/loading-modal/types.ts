import { TransactionMessageType, TransactionMessageText } from '../../core/blockchain/types';

export interface ILoadingModalMessage {
    type: TransactionMessageType;
    text?: TransactionMessageText;
    component?: any; // React.ComponentType<any>;
}

export interface ILoadingModalState {
    isVisible: boolean;
    message: ILoadingModalMessage;
}
