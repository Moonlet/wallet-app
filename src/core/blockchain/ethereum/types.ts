export enum MethodSignature {
    INCREASE_ALLOWANCE = 'increaseAllowance(address,uint256)',
    DELEGATE = 'delegate(address,uint256)',
    UNDELEGATE = 'undelegate(address,uint256)',
    WITHDRAW_DELEGATED = 'withdrawDelegated(address,address)',
    TRANSFER = 'transfer(address,uint256)'
}
