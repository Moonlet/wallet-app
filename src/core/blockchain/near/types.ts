import BigNumber from 'bignumber.js';
import { PosBasicActionType } from '../types/token';

export enum NearTransactionActionType {
    TRANSFER = 'TRANSFER',
    FUNCTION_CALL = 'FUNCTION_CALL'
}

export enum NearFunctionCallMethods {
    DEPOSIT = 'deposit',
    STAKE = 'stake',
    UNSTAKE = 'unstake',
    WITHDRAW = 'withdraw'
}

export interface INearTransactionAction {
    type: NearTransactionActionType;
    params?: any[];
}

export interface INearTransactionAdditionalInfoType {
    currentBlockHash: string;
    actions: INearTransactionAction[];
    posAction?: PosBasicActionType;
    validatorName?: string;
}

export enum NearAccountType {
    DEFAULT = 'DEFAULT',
    CONTRACT = 'CONTRACT'
}

export interface INearAccount {
    address: string;
    name: string;
    amount: BigNumber;
    exists: boolean;
    valid: boolean;
    type?: NearAccountType;
}
