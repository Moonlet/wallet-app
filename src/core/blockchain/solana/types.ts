export enum SolanaTransactionInstructionType {
    TRANSFER = 'TRANSFER',
    CREATE_ACCOUNT_WITH_SEED = 'CREATE_ACCOUNT_WITH_SEED',
    DELEGATE_STAKE = 'DELEGATE_STAKE',
    SPLIT_STAKE = 'SPLIT_STAKE',
    UNSTAKE = 'UNSTAKE'
}

export interface ISolanaTransactionInstruction {
    type: SolanaTransactionInstructionType;
    instruction: any;
}

export interface IStakeAccountFormat {
    index: number;
    validatorId?: string;
    staked?: string;
    unstaked?: string;
    activating?: string;
    deactivating?: string;
}
