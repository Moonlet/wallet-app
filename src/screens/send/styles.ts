import { StyleSheet, Platform } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { BASE_DIMENSION, BORDER_RADIUS } from '../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            paddingBottom: BASE_DIMENSION * 2,
            paddingLeft: BASE_DIMENSION * 2,
            paddingRight: BASE_DIMENSION * 2,
            flexDirection: 'column',
            backgroundColor: theme.colors.appBackground,
            ...StyleSheet.absoluteFillObject
        },
        keyboardAvoidance: {
            flex: 1,
            justifyContent: Platform.OS === 'ios' ? 'space-between' : 'flex-start'
        },
        accountAddress: {
            paddingTop: 40
        },
        address: {
            fontSize: 30,
            textAlign: 'center',
            fontWeight: 'bold',
            marginBottom: 40
        },
        input: {
            flex: 1,
            color: theme.colors.text
        },
        inputAddress: {
            flex: 1,
            color: theme.colors.text,
            paddingRight: BASE_DIMENSION * 2
        },
        receipientLabel: {
            paddingLeft: BASE_DIMENSION,
            marginTop: BASE_DIMENSION * 3,
            color: theme.colors.textSecondary
        },
        receipientNotValid: {
            paddingLeft: BASE_DIMENSION,
            marginBottom: BASE_DIMENSION,
            color: theme.colors.error
        },
        inputBoxAddress: {
            height: 44,
            borderRadius: BORDER_RADIUS,
            borderColor: 'gray',
            alignSelf: 'stretch',
            backgroundColor: theme.colors.inputBackground,
            paddingHorizontal: 12,
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center'
        },
        inputBox: {
            height: BASE_DIMENSION * 5,
            borderRadius: BORDER_RADIUS,
            borderColor: 'gray',
            alignSelf: 'stretch',
            backgroundColor: theme.colors.inputBackground,
            paddingHorizontal: 12,
            marginTop: BASE_DIMENSION,
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center'
        },
        bottom: {
            flex: 1,
            justifyContent: 'flex-end',
            alignItems: 'center',
            marginBottom: BASE_DIMENSION * 2
        },
        bottomButton: {
            width: '90%'
        },
        basicFields: {
            flex: 1,
            paddingTop: BASE_DIMENSION * 5,
            backgroundColor: theme.colors.appBackground
        },
        icon: {
            color: theme.colors.accent,
            marginHorizontal: 0
        },
        rightAddressButton: {
            flexDirection: 'row',
            marginRight: BASE_DIMENSION / 2
        },
        buttonRightOptions: {
            marginTop: BASE_DIMENSION,
            marginBottom: BASE_DIMENSION,
            alignItems: 'flex-end'
        },
        textTranferButton: {
            color: theme.colors.accent,
            lineHeight: 21,
            fontSize: theme.fontSize.small
        }
    });
