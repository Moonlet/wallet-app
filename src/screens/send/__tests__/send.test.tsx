import React from 'react';
import { shallow } from 'enzyme';
import { SendScreenComponent, IProps, navigationOptions } from '../send';
import stylesProvider from '../styles';
import { darkTheme } from '../../../styles/themes/dark-theme';
import { loadTranslations } from '../../../core/i18n';
import BigNumber from 'bignumber.js';
import { Blockchain } from '../../../core/blockchain/types';

jest.mock('../../../redux/config');

const props: IProps = {
    // @ts-ignore
    navigation: {
        navigate: jest.fn()
    },
    styles: stylesProvider(darkTheme),
    theme: darkTheme,
    account: {
        index: 1,
        blockchain: Blockchain.ZILLIQA,
        address: 'zil1vs74hw5k21233h432kj321l3k21b',
        publicKey: '1',
        balance: {
            inProgress: false,
            timestamp: 123,
            value: new BigNumber(12332)
        }
    },
    blockchain: Blockchain.ZILLIQA,
    contacts: {},
    token: {}
};

export default describe('SendScreen', () => {
    beforeAll(async () => {
        await loadTranslations('en');
    });

    // test('renders correctly', () => {
    //     const wrapper = shallow(<SendScreenComponent {...props} />);
    //     expect(wrapper.debug()).toMatchSnapshot();

    //     wrapper.setState({
    //         isValidAddress: true,
    //         amount: '1'
    //     });
    //     expect(wrapper.debug()).toMatchSnapshot();
    // });

    // test('renders correctly - no amount and fee should be present', () => {
    //     const wrapper = shallow(<SendScreenComponent {...props} />);
    //     expect(wrapper.debug()).toMatchSnapshot();
    //     expect(wrapper.debug()).toMatchSnapshot();
    // });

    // test('Confirm button goes on the proper screen', () => {
    //     const wrapper = shallow(<SendScreenComponent {...props} />);
    //     wrapper.setState({
    //         isValidAddress: true,
    //         amount: '1'
    //     });

    //     wrapper.find('[testID="confirm-payment"]').simulate('Press');
    //     expect(props.navigation.navigate).toHaveBeenCalledWith('ConfirmPayment');
    // });

    // test('Button should be disabled if amount is 0', () => {
    //     const wrapper = shallow(<SendScreenComponent {...props} />);
    //     wrapper.setState({
    //         isValidAddress: true
    //     });
    //     expect(wrapper.debug()).toMatchSnapshot();
    // });
    test('Buton should open qr-code', () => {
        const wrapper = shallow(<SendScreenComponent {...props} />);

        wrapper.find('[testID="qrcode-icon"]').simulate('Press');
        expect(wrapper.debug()).toMatchSnapshot();
    });

    test('should change state if address is entered', () => {
        const wrapper: any = shallow(<SendScreenComponent {...props} />).instance();
        wrapper.verifyInputText('address');
        expect(wrapper.state.toAddress).toEqual('address');
    });
    // test('should change state if amount is entered', () => {
    //     const wrapper: any = shallow(<SendScreenComponent {...props} />).instance();
    //     wrapper.addAmount('1');
    //     expect(wrapper.state.amount).toEqual('1');
    // });
    // test('should calculate fee is amount is entered', () => {
    //     const wrapper: any = shallow(<SendScreenComponent {...props} />).instance();
    //     wrapper.addAmount('1');
    //     expect(wrapper.state.fee).toEqual('0.001ZIL');
    // });
    test('should Call verify address', () => {
        const wrapper: any = shallow(<SendScreenComponent {...props} />);
        wrapper.instance().onQrCodeScanned('address');
        expect(wrapper.debug()).toMatchSnapshot();
    });

    // test('onChangeText', () => {
    //     const wrapper: any = shallow(<SendScreenComponent {...props} />);
    //     wrapper.find('[testID="input-address"]').simulate('changeText', 'pass1');
    //     expect(wrapper.debug()).toMatchSnapshot();

    //     wrapper.setState({
    //         isValidAddress: true,
    //         feeOptions: {
    //             feeTotal: new BigNumber(1)
    //         }
    //     });
    //     wrapper.find('[testID="amount"]').simulate('changeText', '10');

    //     expect(wrapper.debug()).toMatchSnapshot();
    // });

    // test('sets correct navigation options', () => {
    //     const navigationProp = {
    //         navigation: { state: { params: { goBack: jest.fn() } } }
    //     };
    //     const options = navigationOptions(navigationProp);
    //     expect(options).toMatchSnapshot();
    // });
});
