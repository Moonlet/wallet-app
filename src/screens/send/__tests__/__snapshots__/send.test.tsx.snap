// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendScreen Buton should open qr-code 1`] = `
"<View style={{...}}>
  <withTheme(TextComponent) style={{...}}>
    Reusable component
  </withTheme(TextComponent)>
  <View style={{...}}>
    <TextInput testID=\\"input-address\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Search public addresses or domain names\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    <TouchableOpacity testID=\\"qrcode-icon\\" onPress={[Function: _callee]} style={{...}} activeOpacity={0.2}>
      <Icon name=\\"qr-code-scan\\" size={20} style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
  <QrModalReaderComponent onQrCodeScanned={[Function]} />
</View>"
`;

exports[`SendScreen Button should be disabled if amount is 0 1`] = `
"<View style={{...}}>
  <withTheme(TextComponent) style={{...}}>
    Reusable component
  </withTheme(TextComponent)>
  <View style={{...}}>
    <TextInput testID=\\"input-address\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Search public addresses or domain names\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    <TouchableOpacity testID=\\"qrcode-icon\\" onPress={[Function: _callee]} style={{...}} activeOpacity={0.2}>
      <Icon name=\\"qr-code-scan\\" size={20} style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
  <View style={{...}}>
    <View style={{...}}>
      <TextInput testID=\\"amount\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Amount\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    </View>
    <View style={{...}}>
      <TextInput testID=\\"fee\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" editable={false} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    </View>
    <View style={{...}}>
      <withTheme(ButtonComponent) testID=\\"confirm-payment\\" style={{...}} primary={true} disabled={true} onPress={[Function: onPress]}>
        Confirm payment
      </withTheme(ButtonComponent)>
    </View>
  </View>
  <QrModalReaderComponent onQrCodeScanned={[Function]} />
</View>"
`;

exports[`SendScreen onChangeText 1`] = `
"<View style={{...}}>
  <withTheme(TextComponent) style={{...}}>
    Reusable component
  </withTheme(TextComponent)>
  <withTheme(TextComponent) style={{...}}>
    Recipient
  </withTheme(TextComponent)>
  <View style={{...}}>
    <TextInput testID=\\"input-address\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Search public addresses or domain names\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"pass1\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    <TouchableOpacity testID=\\"qrcode-icon\\" onPress={[Function: _callee]} style={{...}} activeOpacity={0.2}>
      <Icon name=\\"qr-code-scan\\" size={20} style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
  <QrModalReaderComponent onQrCodeScanned={[Function]} />
</View>"
`;

exports[`SendScreen onChangeText 2`] = `
"<View style={{...}}>
  <withTheme(TextComponent) style={{...}}>
    Reusable component
  </withTheme(TextComponent)>
  <withTheme(TextComponent) style={{...}}>
    Recipient
  </withTheme(TextComponent)>
  <View style={{...}}>
    <TextInput testID=\\"input-address\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Search public addresses or domain names\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"pass1\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    <TouchableOpacity testID=\\"qrcode-icon\\" onPress={[Function: _callee]} style={{...}} activeOpacity={0.2}>
      <Icon name=\\"qr-code-scan\\" size={20} style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
  <View style={{...}}>
    <View style={{...}}>
      <TextInput testID=\\"amount\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Amount\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"10\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    </View>
    <View style={{...}}>
      <TextInput testID=\\"fee\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"0.001ZIL\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" editable={false} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    </View>
    <View style={{...}}>
      <withTheme(ButtonComponent) testID=\\"confirm-payment\\" style={{...}} primary={true} disabled={false} onPress={[Function: onPress]}>
        Confirm payment
      </withTheme(ButtonComponent)>
    </View>
  </View>
  <QrModalReaderComponent onQrCodeScanned={[Function]} />
</View>"
`;

exports[`SendScreen renders correctly - no amount and fee should be present 1`] = `
"<View style={{...}}>
  <withTheme(TextComponent) style={{...}}>
    Reusable component
  </withTheme(TextComponent)>
  <View style={{...}}>
    <TextInput testID=\\"input-address\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Search public addresses or domain names\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    <TouchableOpacity testID=\\"qrcode-icon\\" onPress={[Function: _callee]} style={{...}} activeOpacity={0.2}>
      <Icon name=\\"qr-code-scan\\" size={20} style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
  <QrModalReaderComponent onQrCodeScanned={[Function]} />
</View>"
`;

exports[`SendScreen renders correctly - no amount and fee should be present 2`] = `
"<View style={{...}}>
  <withTheme(TextComponent) style={{...}}>
    Reusable component
  </withTheme(TextComponent)>
  <View style={{...}}>
    <TextInput testID=\\"input-address\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Search public addresses or domain names\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    <TouchableOpacity testID=\\"qrcode-icon\\" onPress={[Function: _callee]} style={{...}} activeOpacity={0.2}>
      <Icon name=\\"qr-code-scan\\" size={20} style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
  <QrModalReaderComponent onQrCodeScanned={[Function]} />
</View>"
`;

exports[`SendScreen renders correctly 1`] = `
"<View style={{...}}>
  <withTheme(TextComponent) style={{...}}>
    Reusable component
  </withTheme(TextComponent)>
  <View style={{...}}>
    <TextInput testID=\\"input-address\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Search public addresses or domain names\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    <TouchableOpacity testID=\\"qrcode-icon\\" onPress={[Function: _callee]} style={{...}} activeOpacity={0.2}>
      <Icon name=\\"qr-code-scan\\" size={20} style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
  <QrModalReaderComponent onQrCodeScanned={[Function]} />
</View>"
`;

exports[`SendScreen renders correctly 2`] = `
"<View style={{...}}>
  <withTheme(TextComponent) style={{...}}>
    Reusable component
  </withTheme(TextComponent)>
  <View style={{...}}>
    <TextInput testID=\\"input-address\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Search public addresses or domain names\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    <TouchableOpacity testID=\\"qrcode-icon\\" onPress={[Function: _callee]} style={{...}} activeOpacity={0.2}>
      <Icon name=\\"qr-code-scan\\" size={20} style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
  <View style={{...}}>
    <View style={{...}}>
      <TextInput testID=\\"amount\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Amount\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"1\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    </View>
    <View style={{...}}>
      <TextInput testID=\\"fee\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" editable={false} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    </View>
    <View style={{...}}>
      <withTheme(ButtonComponent) testID=\\"confirm-payment\\" style={{...}} primary={true} disabled={false} onPress={[Function: onPress]}>
        Confirm payment
      </withTheme(ButtonComponent)>
    </View>
  </View>
  <QrModalReaderComponent onQrCodeScanned={[Function]} />
</View>"
`;

exports[`SendScreen sets correct navigation options 1`] = `
Object {
  "headerLeft": [Function],
  "title": "Send",
}
`;

exports[`SendScreen sets correct navigation options 2`] = `
<withTheme(HeaderLeftComponent)
  icon="close"
  onPress={[Function]}
  text="Close"
/>
`;

exports[`SendScreen should Call verify address 1`] = `
"<View style={{...}}>
  <withTheme(TextComponent) style={{...}}>
    Reusable component
  </withTheme(TextComponent)>
  <withTheme(TextComponent) style={{...}}>
    Recipient
  </withTheme(TextComponent)>
  <View style={{...}}>
    <TextInput testID=\\"input-address\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"Search public addresses or domain names\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" value=\\"address\\" onChangeText={[Function: onChangeText]} allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
    <TouchableOpacity testID=\\"qrcode-icon\\" onPress={[Function: _callee]} style={{...}} activeOpacity={0.2}>
      <Icon name=\\"qr-code-scan\\" size={20} style={{...}} allowFontScaling={false} />
    </TouchableOpacity>
  </View>
  <QrModalReaderComponent onQrCodeScanned={[Function]} />
</View>"
`;
